api/v1alpha1/api.go

```diff
@@ -61,12 +61,21 @@ type AIGatewayRouteList struct {
 }
 
 // AIGatewayRouteSpec details the AIGatewayRoute configuration.
+//
+// +kubebuilder:validation:XValidation:rule="!has(self.parentRefs) || !has(self.targetRefs) || size(self.targetRefs) == 0", message="targetRefs is deprecated, use parentRefs only"
 type AIGatewayRouteSpec struct {
 	// TargetRefs are the names of the Gateway resources this AIGatewayRoute is being attached to.
 	//
-	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=128
-	TargetRefs []gwapiv1a2.LocalPolicyTargetReferenceWithSectionName `json:"targetRefs"`
+	// Deprecated: use the ParentRefs field instead. This field will be dropped in Envoy AI Gateway v0.3.0.
+	TargetRefs []gwapiv1a2.LocalPolicyTargetReferenceWithSectionName `json:"targetRefs,omitempty"`
+	// ParentRefs are the names of the Gateway resources this AIGatewayRoute is being attached to.
+	// Cross namespace references are not supported. In other words, the Gateway resources must be in the
```

Comment:
can this be relaxed in the future / follow up PR ?
its very common for the Gateway and Route to live in different ns

Reply:
Yeah possible